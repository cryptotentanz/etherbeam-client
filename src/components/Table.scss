@use '../styles/resources/sizes';
@use '../styles/resources/colors';

.my-table {
  border-collapse: collapse;
  border-style: solid;
  border-width: sizes.$border;
  width: 100%;
  padding: sizes.$spacer;

  th {
    letter-spacing: sizes.$letter-spacing;
  }

  th,
  td {
    border-style: solid;
    border-width: sizes.$border;
    padding: sizes.$spacer-s;
  }

  // &-primary {
  //   background-color: colors.$bg-primary;
  //   border-color: colors.$bd-primary;
  //   color: colors.$fg-light;

  // }

  // &-secondary {
  //   background-color: colors.$bg-secondary;
  //   border-color: colors.$bd-secondary;
  //   color: colors.$fg-light;

  //   th,
  //   td {
  //     border-color: colors.$bd-secondary;
  //   }
  // }

  // &-dark {
  //   background-color: colors.$bg-dark;
  //   border-color: colors.$bd-dark;
  //   color: colors.$fg-dark;

  //   th,
  //   td {
  //     border-color: colors.$bd-dark;
  //   }
  // }

  // &-light {
  //   background-color: colors.$bg-light;
  //   border-color: colors.$bd-light;
  //   color: colors.$fg-light;

  //   th,
  //   td {
  //     border-color: colors.$bd-light;
  //   }
  // }

  // &-positive {
  //   background-color: colors.$bg-positive;
  //   border-color: colors.$bd-positive;
  //   color: colors.$fg-dark;

  //   th,
  //   td {
  //     border-color: colors.$bd-positive;
  //   }
  // }

  // &-negative {
  //   background-color: colors.$bg-negative;
  //   border-color: colors.$bd-negative;
  //   color: colors.$fg-dark;

  //   th,
  //   td {
  //     border-color: colors.$bd-negative;
  //   }
  // }
}
