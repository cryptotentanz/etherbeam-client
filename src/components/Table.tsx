import React, { FC, ReactNode } from 'react'
import './Table.scss'

type ColorProp = 'dark' | 'light' | 'primary' | 'secondary' | 'positive' | 'negative'

interface TableProps {
  color?: ColorProp
  children?: ReactNode
}

const Table: FC<TableProps> = ({ color = 'dark', children }: TableProps) => {
  return <table className={`my-table my-${color}-bg`}>{children}</table>
}

export default Table
