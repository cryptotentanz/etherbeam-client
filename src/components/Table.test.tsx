import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Table from './Table'

describe('<Table />', () => {
  const content = 'Content'

  let component: ShallowWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = shallow(<Table>{content}</Table>)
    })

    it('should render', () => {
      const table = component.find('table')
      expect(table.prop('children')).toBe(content)
      expect(table.hasClass('my-table')).toBeTruthy()
      expect(table.hasClass('my-dark-bg')).toBeTruthy()
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('Color', () => {
    beforeAll(() => {
      component = shallow(<Table color="primary">{content}</Table>)
    })

    it('should render', () => expect(component.find('table').hasClass('my-primary-bg')).toBeTruthy())
  })
})
