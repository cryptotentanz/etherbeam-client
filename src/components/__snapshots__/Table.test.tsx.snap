// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> Default should match snapshot 1`] = `
Array [
  "Content",
  <table
    class="my-table my-dark-bg"
  />,
]
`;
